browser/ts/index.ts

```diff
@@ -0,0 +1,370 @@
+import { SignedOrder } from '@0x/order-utils';
+import { wasmBuffer } from './generated/wasm_buffer';
+import { BigNumber } from '@0x/utils';
+
+export { SignedOrder } from '@0x/order-utils';
+export { BigNumber } from '@0x/utils';
+
+// Side-effect import
+// Side-effects include adding an `fs` and `Go` property on the global object.
+import './wasm_exec';
+
+// The Go code sets certain global values and this is our only way of
+// interacting with it. Define those values and their types here.
+declare global {
+    // Defined in wasm_exec.ts
+    class Go {
+        run(instance: WebAssembly.Instance): void;
+        importObject: any;
+    }
+
+    // Defined in ../go/main.go
+    const zeroExMesh: ZeroExMesh;
+}
+
+/**
+ * A set of configuration options for Mesh.
+ */
+export interface Config {
+    // The URL of an Ethereum node which supports the Ethereum JSON RPC API.
+    // Used to validate and watch orders.
+    ethereumRPCURL: string;
+    // The network ID to use when communicating with Ethereum.
+    ethereumNetworkID: number;
+    // UseBootstrapList is whether to use the predetermined list of peers to
+    // bootstrap the DHT and peer discovery. Defaults to true.
+    useBootstrapList?: boolean;
+    // The amount of time (in seconds) before the order's stipulated expiration
+    // time that it will be considered expired. Higher values will cause orders
+    // to be considered invalid sooner. Defaluts to 10.
+    orderExpirationBufferSeconds?: number;
+    // The polling interval (in seconds) to wait before checking for a new
+    // Ethereum block that might contain transactions that impact the
+    // fillability of orders stored by Mesh. Different networks have different
+    // block producing intervals: POW networks are typically slower (e.g.,
+    // Mainnet) and POA networks faster (e.g., Kovan) so one should adjust the
+    // polling interval accordingly. Defaults to 5.
+    blockPollingIntervalSeconds?: number;
+    // The maximum request Content-Length accepted by the backing Ethereum RPC
+    // endpoint used by Mesh. Geth & Infura both limit a request's content
+    // length to 1024 * 512 Bytes. Parity and Alchemy have much higher limits.
+    // When batch validating 0x orders, we will fit as many orders into a
+    // request without crossing the max content length. The default value is
+    // appropriate for operators using Geth or Infura. If using Alchemy or
+    // Parity, feel free to double the default max in order to reduce the number
+    // of RPC calls made by Mesh. Defaults to 524288 bytes.
+    ethereumRPCMaxContentLength?: number;
```

Comment:
How will we remember to update these in both places? Can we perhaps add a comment/warning in both places mentioning the need to edit the other?

Reply:
I'll add a comment for now. We can potentially generate code in both places from a canonical source (e.g. JSON) in the future.